### ISSUE TITLE
Cannot change JS code location and run on iPhone locally

### ISSUE THREAD

**Issue by amloelxer** [electrode-io/electrode-native#688]
When using `ern run-ios` and building in debug mode on the actual iPhone, I cannot connect to the development server. This is because my phone cannot access `localhost:8081` even though it is on the same network as my computer. I know this is a common problem in "vanilla" react-native, but the solution there is to just change the jsCodeLocation in the app delegate (usually to your computers IP) and everything usually works fine. In Electrode native, this isn't the case because I believe `ElectrodeReactNative.m` is generated every time `ern run-ios` is ran and the URL is set there. 
```
NSURL *url;
        if (self.debugEnabled) {
            url = [NSURL URLWithString:@"http://localhost:8081/index.ios.bundle?platform=ios&dev=true"];
            NSLog(@"using local port to debug");
        } else {
            url = [self allJSBundleFiles][0];
        }  
```
etc etc.
Is there anyway to change the JS url dynamically? Or any way this could be a possible feature in the future by adding a flag to `ern-run ios`? Everything works fine if I do `ern run-ios --debug=false` but I would like to try get debug mode to run on my phone.

**Comment by belemaire** [comment]
@amloelxer 
Have you actually tried modifying the location manually from `localhost` to your current computers IP? Even though it will get overwritten on next generation as you are mentioning, but just to know if it would actually work this way ?
Then indeed you are correct, we could add a flag `--packager-ip` for example to `run-ios` in case you want to use something different than `localhost`. That would be a good idea as in theory a packager could be run on any ip (and even port, so we would also probably add `--packager-port` in addition while at it). We would need to do it both for Android and iOS for consistency, even though it is less problematic on Android, it is still a nice feature to have to offer more flexibility.

**Comment by amloelxer** [comment]
@belemaire it works if you edit it manually and then build from the Xcode. As a result this command improvement is much less of a want for me, but would definitely be a nice to have. I think in a perfect world I would like to not open Xcode at all when running `ern run-ios`, but realize that isn't super feasible. Also I'm not sure if my frequency of using `ern run-ios` is atypical, but definitely food for thought. Thanks for being responsive as always!

**Comment by belemaire** [comment]
Sure, our iOS engineer is coming back on Monday, so we might look into it together. Setting the milestone on this one for v0.16.0 (March 26th) as I think it would be a nice to have in term of feature and also that it shouldn't be too difficult to implement.
@clairewli let's discuss this on Monday when you are back :)
@amloelxer will keep you posted

**Comment by belemaire** [comment]
@clairewli 
I have updated `run-ios` to accept `--host` and `--port` extra options (defaults to `localhost` and `8081`) as of https://github.com/electrode-io/electrode-native/commit/b7e4c98c58e042d73233ae2d968c265c981bc168. The value of these options are now being stored in the Runner configuration file (`RunnerConfig.m`), in addition to the already existing ones, as follow :

```
NSString *const MainMiniAppName = @"testMiniApp";
BOOL const RnDevSupportEnabled = YES;
NSString *const PackagerHost = @"localhost";
NSString *const PackagerPort = @"8081";
```

Idea is now to update iOS Container so that the configuration that is passed to initialize container will now also include the packager host/port, and will properly use these values to build the url instead of hardcoding it to `localhost:8081` at this LOC :
https://github.com/electrode-io/electrode-native/blob/d70019535178b1ca2f95088e1332a52acce1bcb0/ern-container-gen/src/generators/ios/hull/ElectrodeContainer/ElectrodeReactNative.m#L54

Runner hull will also have to be updated to pass the PackagerHost/PackagerPort from the RunnerConfig to the Container initialization.

Please let me know if you have any questions.

**Comment by belemaire** [comment]
@clairewli Moving this one from v0.16.0 to v0.17.0 (next) release

**Comment by adamweeks** [comment]
Hey team, I see that this was no longer implemented in v0.17. Do you have a workaround for us on a corporate laptop that will not allow port 8081? 

I've tried modifying [this](https://github.com/electrode-io/electrode-native/blob/master/ern-container-gen/src/generators/ios/hull/ElectrodeContainer/ElectrodeReactNative.m#L54) and using the `--port` command to match, but it still fails on the last step with the following error message:

```
✖ XCode xcbuild command failed with exit code 65
(node:70292) UnhandledPromiseRejectionWarning: Error: XCode xcbuild command failed with exit code 65
    at ChildProcess.xcodebuildProc.on.code (/Users/adweeks/.ern/versions/0.18.2/node_modules/ern-local-cli/src/lib/utils.ts:969:13)
    at emitTwo (events.js:126:13)
    at ChildProcess.emit (events.js:214:7)
    at maybeClose (internal/child_process.js:925:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:209:5)
(node:70292) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:70292) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Thanks for any help you can provide @belemaire (it this isn't related to this issue, I can create a new one).

**Comment by belemaire** [comment]
Moving this one to our backlog
